### PR TITLE
feat(sdk): add retry policy support to kfp v2

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Adds retry policy support to KFP v2 SDK.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title 
convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/compiler/compiler_test.py [hunk 1]
```diff
@@ -1048,5 +1048,38 @@ def load_compiled_file(filename: str) -> Dict[str, Any]:
...
+            """Hello world component."""
+            return text
+
+        @dsl.pipeline(name='hello-world', description='A simple intro pipeline')
+        def pipeline_hello_world(text: str = 'hi there'):
+            """Hello world pipeline."""
+
+            hello_world(text=text).set_retry(
+                num_retries=3,
+                backoff_duration='30s',
```
[the remark this answers is not in the document]
Comment (chensun) [reply sdk/python/kfp/compiler/compiler_test.py:1066]: on a second look, it's fine to keep this as-is, as long as the samples under `test_data/pipelines` doesn't show these no-op usage.

**Review by chensun — COMMENTED** [review]
/lgtm
/approve

Thanks!

**Comment by connor-mccarthy** [comment]
@chensun, can you re-LGTM after rebase and one-character cleanup commit?

**Review by chensun — APPROVED** [review]
/lgtm

**Pull request merged**